### PR TITLE
Feature/delete

### PR DESCRIPTION
**Pull request by Steffen-Anslinger**

### CODE REVIEW

File: components/DetailsPage/index.js [hunk 1]
```diff
@@ -60,14 +69,18 @@ export default function DetailsPage({
...
+          <ProjectDetails
             project={project}
             onToggleFavourite={onToggleFavourite}
             isFavourite={isFavourite}
             favourites={favourites}
           />
           <StyledLink href="/">Back</StyledLink>
           <button onClick={() => setEditMode(true)}>Edit</button>
+
+          <button type="button" onClick={handleSubmit(handleDeleteProject)}>
```
[the remark this answers is not in the document]
Comment (Roland-Hufnagel) [reply components/DetailsPage/index.js:81]: And I am wondering, why do you use the useForm hook here and passing the props to the EditForm Component. I think a better way is to use the hook inside that component.

File: components/DetailsPage/index.js [hunk 2]
```diff
@@ -41,6 +40,16 @@ export default function DetailsPage({
   if (!project) {
     return <h1>Project not found</h1>;
   }
+
+  async function handleDeleteProject() {
+    if (window.confirm("Are you sure you want to delete this project?")) {
+      await fetch(`/api/projects/${id}`, { method: "DELETE" });
+      const response = await fetch("/api/projects");
+      mutate("/api/projects", response.json());
```
[the remark this answers is not in the document]
Comment (Roland-Hufnagel) [reply components/DetailsPage/index.js:48]: I recommend a bit of error handling. What if deleting is not possible? You should at least test the `response.ok` value.

File: pages/api/projects/[id].js [hunk 3]
```diff
@@ -20,4 +20,9 @@ export default async function handler(request, response) {
     await Project.findByIdAndUpdate(id, updatedProject);
     response.status(200).json({ status: `Project successfully updated!` });
   }
+
+  if (request.method === "DELETE") {
+    await Project.findByIdAndDelete(id);
```
Comment (Roland-Hufnagel) [inline pages/api/projects/[id].js:25]: You should implement some error handling here as well.

**Review by Roland-Hufnagel — CHANGES_REQUESTED** [review]
Hey team. You did a lot of good work so far. 💪

It took some time to get familiar with your code and I was struggling with some structure and names. I recommend not to name the component 'DetailsPage'. I expect a 'page' to be in the pages folder. Like you do the import: `import ProjectDetails from "@/components/DetailsPage";` it's a good idea to name the component 'ProjectDetails'. 

You do a lot of work inside this component. The better way is to use the router and do the fetch in the page (pages/projects/[id].js). 

I see a navigation component on every page. Thats really redundant. I think the navigation should be on every page, or not? How can you apply this?

Take a look at the comments and try to improve your code a bit.

File: components/EditForm/index.js [hunk 4]
```diff
@@ -1,77 +1,98 @@
...
+
+  const onSubmit = async (formData) => {
+    await fetch(`/api/projects/${id}`, {
+      method: "PUT",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(formData),
+    });
+    const response = await fetch(`/api/projects/${id}`);
```
Comment (moonwave99) [inline components/EditForm/index.js:20]: Why the second fetch?

**Review by moonwave99 — CHANGES_REQUESTED** [review]

File: components/EditForm/index.js [hunk 5]
```diff
@@ -1,77 +1,98 @@
...
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(formData),
+    });
+
+    if (response.ok) {
+      setEditMode(false);
+      mutate(`/api/projects/${id}`);
+    }
```
Comment (Roland-Hufnagel) [inline components/EditForm/index.js:24]: What happens if the response is not ok?

File: pages/api/projects/[id].js [hunk 6]
```diff
@@ -20,4 +20,14 @@ export default async function handler(request, response) {
     await Project.findByIdAndUpdate(id, updatedProject);
     response.status(200).json({ status: `Project successfully updated!` });
   }
+
+  if (request.method === "DELETE") {
+    const deletedProject = await Project.findByIdAndDelete(id);
+    if (deletedProject) {
+      response.status(200).json({ status: `Project successfully deleted.` });
+    }
+    if (!deletedProject) {
```
Comment (Roland-Hufnagel) [inline pages/api/projects/[id].js:29]: `else` would be fine

**Review by Roland-Hufnagel — APPROVED** [review]
This looks fine now. Good refactoring. 💪I just mentioned two small things.

**Pull request merged**